### PR TITLE
[TierTwo] Unify MNs info gathering into a single getMNsInfo() call

### PR DESCRIPTION
**Pull request by furszy**
This simplifies the 'getmasternodecount' workflow that was walking through the 'mapMasternodes' 3 times locking the manager mutex when it can gather the information looping over it just once.

Plus, added a second commit cleaning the `HasZerocoinPublicSpendInputs` unused function and `ScoreCompare` unused class.

### CODE REVIEW

File: src/primitives/transaction.cpp [hunk 1]
```diff
@@ -183,15 +173,15 @@ bool CTransaction::HasP2CSOutputs() const
 CAmount CTransaction::GetValueOut() const
 {
     CAmount nValueOut = 0;
-    for (std::vector<CTxOut>::const_iterator it(vout.begin()); it != vout.end(); ++it) {
+    for (const CTxOut& it : vout) {
```
Comment (random-zebra) [inline src/primitives/transaction.cpp:176]: μ-nit: better to call it `out` rather than `it` as it's no longer an iterator.

**Review by random-zebra — APPROVED** [review]
utACK eda79c7aed7657766b94ba18c8a8d3bb5ca0bce3

File: src/primitives/transaction.cpp (hunk 1 above)
Comment (furszy) [reply src/primitives/transaction.cpp:176]: sure, one sec

**Comment by furszy** [comment]
pushed μ-nit

**Pull request merged**